datafusion/expr/src/expr.rs

```diff
@@ -1133,14 +1133,23 @@ impl Expr {
     }
 
     /// Return `self AS name` alias expression
+    /// Removes existing Alias if exists
+    ///
+    /// # Example
+    /// ```
+    /// # use datafusion_expr::col;
+    /// // `foo as "bar" as "baz"` is resolved to `foo as "baz"`
+    /// let expr = col("foo").alias("bar").alias("baz");
+    /// assert_eq!(expr, col("foo").alias("baz"));
+    /// ```
     pub fn alias(self, name: impl Into<String>) -> Expr {
         match self {
             Expr::Sort(Sort {
                 expr,
                 asc,
                 nulls_first,
             }) => Expr::Sort(Sort::new(Box::new(expr.alias(name)), asc, nulls_first)),
-            _ => Expr::Alias(Alias::new(self, None::<&str>, name.into())),
+            _ => Expr::Alias(Alias::new(self.unalias(), None::<&str>, name.into())),
```

Comment:
Is changing how `alias()` works required for this PR?
I think the change make sense but extracting it to a separate PR might be better...

Reply:
It ties into why the cse was changed, i.e.. removing extra aliases.

It's a tiny change that I don't think deserves a separate PR.

Reply:
Yeah a tiny one, but technically it's an API behaviour change.

Reply:
It's not a breaking change, but I'm fine with deferring it.